apis/infrastructure/v1beta1/byohost_webhook.go

```diff
@@ -20,23 +22,55 @@ type ByoHostValidator struct {
 	decoder *admission.Decoder
 }
 
+// To allow byoh manager service account to patch ByoHost CR
+const managerServiceAccount = "system:serviceaccount:byoh-system:byoh-controller-manager"
+
 // nolint: gocritic
 // Handle handles all the requests for ByoHost resource
 func (v *ByoHostValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation == v1.Delete {
-		byoHost := &ByoHost{}
-		err := v.decoder.DecodeRaw(req.OldObject, byoHost)
-		if err != nil {
-			return admission.Errored(http.StatusBadRequest, err)
-		}
-
-		if byoHost.Status.MachineRef != nil {
-			return admission.Denied("cannot delete ByoHost when MachineRef is assigned")
-		}
+	var response admission.Response
+
+	switch req.Operation {
+	case v1.Create, v1.Update:
+		response = v.handleCreateUpdate(&req)
+	case v1.Delete:
+		response = v.handleDelete(&req)
+	default:
+		response = admission.Allowed("")
 	}
+	return response
+}
 
-	// TODO: verify if req.UserInfo.Username has rbac permission to update the byohost
+func (v *ByoHostValidator) handleCreateUpdate(req *admission.Request) admission.Response {
```

Comment:
Can we create two separate functions `handleCreate` and `handleUpdate` and then call `handleCreate` inside `handleUpdate` put the if clause meant for update case in the same func, such that we don't mix two functionalities in one function, wdyt?

Reply:
Yes, that was my initial implementation but as some of the code was going to be common, I kept it like this no strong opinions though. With the method name `handleCreateUpdate`, I guess it's okay from the readability perspective and code is also not very complicated at the moment.

>and then call handleCreate inside handleUpdate

This will bring some indirection which might not be very readable as to why we are calling handleCreate.